src/libraries/System.Net.Quic/src/System/Net/Quic/Internal/MsQuicConfiguration.cs

```diff
@@ -171,11 +171,14 @@ private static unsafe MsQuicSafeHandle Create(QuicConnectionOptions options, QUI
 
                 byte[] certificateData;
 
-                if (intermediates?.Length > 0)
+                if (intermediates.Length > 0)
                 {
                     X509Certificate2Collection collection = new X509Certificate2Collection();
                     collection.Add(certificate);
-                    collection.AddRange(intermediates);
+                    foreach (X509Certificate2 ca in intermediates)
+                    {
+                        collection.Add(ca);
+                    }
```

Comment:
This could be slower if we expected a lot of certs.  I assume we don't?

Reply:
typically this would be 1-3 items. I was looking at https://github.com/dotnet/runtime/issues/1530 but it did not seems like I can use the `AddRage` for generic collection. We actually allocate it just to make the export easy e.g. produce the `Pkcs12` blob. If needed we may suck in some crypto primitives directly.

Reply:
Should be fine as you have it.

Reply:
I'm not sure that would make measurable improvement. There was some discussion on the issue linked above.
If anything I would push to avoid allocation of  the collection.